examples/plot_post_stack_inversion.py

```diff
@@ -0,0 +1,112 @@
+"""
+Post Stack Inversion - 3D
+=========================
+This illustration demonstrates the implementation of a distributed 3D Post-stack inversion. It involves
+modelling a 3-D synthetic post-stack seismic data from a profile of the subsurface acoustic impedence.
+"""
+
+import numpy as np
+from scipy.signal import filtfilt
+from pylops import ricker, PoststackLinearModelling
+from matplotlib import pyplot as plt
+from mpi4py import MPI
+from pylops_mpi import MPIBlockDiag, DistributedArray
+
+plt.close("all")
+rank = MPI.COMM_WORLD.Get_rank()
+size = MPI.COMM_WORLD.Get_size()
+
+###############################################################################
+# Let's start by defining the important parameters required to model the
+# ``pylops.avo.poststack.PoststackLinearModelling`` operator.
+
+# Model
+model = np.load("../testdata/avo/poststack_model.npz")
+x, z, m = model['x'], model['z'], np.log(model['model'])
+
+# Making m a 3-D
+ny_i = 10
+y = np.arange(ny_i)
+m3d_i = np.tile(m[:, :, np.newaxis], (1, 1, ny_i)).transpose((2, 1, 0))
+ny_i, nx, nz = m3d_i.shape
+
+# Size of y at all ranks
+ny = MPI.COMM_WORLD.allreduce(ny_i)
+
+# Smooth model
+nsmoothz, nsmoothx = 30, 20
+mback = filtfilt(np.ones(nsmoothz) / float(nsmoothz), 1, m, axis=0)
+mback = filtfilt(np.ones(nsmoothx) / float(nsmoothx), 1, mback, axis=1)
+# Making mback a 3-D
+mback3d_i = np.tile(mback[:, :, np.newaxis], (1, 1, ny_i)).transpose((2, 1, 0))
+
+# wavelet
+dt = 0.004
+t0 = np.arange(nz) * dt
+ntwav = 41
+wav = ricker(t0[:ntwav // 2 + 1], 15)[0]
+
+# Collecting all the m3d and mback3d at all ranks
+m3d = np.concatenate(MPI.COMM_WORLD.allgather(m3d_i))
+mback3d = np.concatenate(MPI.COMM_WORLD.allgather(mback3d_i))
+
+###############################################################################
+# We model the data using both the dense and linear operator version of
+# ``pylops.avo.poststack.PostStackLinearModelling`` at each rank and pass
+# these operators to the ``pylops_mpi.MPIBlockDiag`` operator. Furthermore,
+# we use this MPIBlockDiag to perform forward operation on each operator
+# at different ranks to get the ``data``.
+
+# Flattening modelling data
+m3d_dist = DistributedArray(global_shape=nx * ny * nz)
+m3d_dist[:] = m3d_i.transpose((2, 0, 1)).flatten()
+
+# Linear PostStackLinearModelling
+PPop = PoststackLinearModelling(wav, nt0=nz, spatdims=(ny_i, nx))
+BDiag = MPIBlockDiag(ops=[PPop, ])
+
+# Data
+d_dist = BDiag * m3d_dist
```

Comment:
always better to use @ instead of * (I know sometimes we still have some * in pylops here and there...)

Reply:
Yup noted will use @ everywhere from now onwards